src/integTest/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowPanelTest.kt

```diff
@@ -0,0 +1,104 @@
+import snyk.iac.ui.toolwindow.IacIssueTreeNode
+import javax.swing.tree.TreeNode
+
+class SnykToolWindowPanelTest : HeavyPlatformTestCase() {
+
+    private val iacGoofJson = getResourceAsString("iac-test-results/infrastructure-as-code-goof.json")
+
+    private fun getResourceAsString(resourceName: String): String = javaClass.classLoader
+        .getResource(resourceName)!!.readText(Charsets.UTF_8)
+
+    override fun setUp() {
+        super.setUp()
+        unmockkAll()
+        resetSettings(project)
+        setupDummyCliFile()
+    }
+
+    override fun tearDown() {
+        unmockkAll()
+        resetSettings(project)
+        removeDummyCliFile()
+        super.tearDown()
+    }
+
+    @Test
+    fun testSeverityFilterForIacResult() {
+        // pre-test setup
+        val settings = pluginSettings()
+        settings.ossScanEnable = false
+        settings.snykCodeSecurityIssuesScanEnable = false
+        settings.snykCodeQualityIssuesScanEnable = false
+        settings.iacScanEnabled = true
+
+        val toolWindowPanel = project.service<SnykToolWindowPanel>()
+
+        val isIacEnabledRegistryValue = Registry.get("snyk.preview.iac.enabled")
+        val isIacEnabledOldValue = isIacEnabledRegistryValue.asBoolean()
+        isIacEnabledRegistryValue.setValue(true)
+
+        // mock IaC results
+        val mockRunner = mockk<ConsoleCommandRunner>()
+        every {
+            mockRunner.execute(
+                listOf(getCliFile().absolutePath, "iac", "test", "--json"),
+                project.basePath!!,
+                project = project
+            )
+        } returns (iacGoofJson)
+
+        getIacService(project).setConsoleCommandRunner(mockRunner)
+
+        // actual test run
+
+        project.service<SnykTaskQueueService>().scan()
+
+        PlatformTestUtil.waitWhileBusy(toolWindowPanel.getTree())
+
+        val rootIacIssuesTreeNode = toolWindowPanel.getRootIacIssuesTreeNode()
+        fun isMediumSeverityShown(): Boolean = rootIacIssuesTreeNode.children().asSequence()
+            .flatMap { (it as TreeNode).children().asSequence() }
+            .any {
+                it is IacIssueTreeNode &&
+                    it.userObject is IacIssue &&
+                    (it.userObject as IacIssue).severity == Severity.MEDIUM
+            }
+
+        assertTrue("Medium severity IaC results should be shown by default", isMediumSeverityShown())
+
+        val mediumSeverityFilterAction =
+            ActionManager.getInstance().getAction("io.snyk.plugin.ui.actions.SnykTreeMediumSeverityFilterAction")
+                as SnykTreeMediumSeverityFilterAction
+        mediumSeverityFilterAction.setSelected(TestActionEvent(), false)
+
+        PlatformTestUtil.waitWhileBusy(toolWindowPanel.getTree())
+
+        assertFalse("Medium severity IaC results should NOT be shown after filtering", isMediumSeverityShown())
+
+        // restore modified Registry value
+        isIacEnabledRegistryValue.setValue(isIacEnabledOldValue)
```

Comment:
should this be in setup/teardown?

Reply:
That registry settings is changed explicitly in that test only. So I think we should revert it back here explicitly too and not in tearDown which executed for all (future) tests in that class.
If we later find the way to reset the whole App state for each test to it initial state, that would solve the problem and no such revert setting will be needed.

Reply:
Ok, sounds like a plan. Hoping for resetting the whole app state. Else it would be cleaner to have setup/teardown ensure that the expected pre-condition is met, without a test having to cleanup after itself (jumping out of a test with an exception has happened when debugging to me).

Reply:
I'm afraid resetting the whole Registry might be out of our control as it's IntelliJ own class....